cinema/serializers.py

```diff
@@ -68,13 +77,58 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
+class TicketSeatRowSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketSeatRowSerializer(
+        many=True,
+        read_only=True,
+        source="tickets",
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "movie_session", "row", "seat")
+
+    def validate(self, data):
+        data = super(TicketSerializer).validate(data)
+        ticket = Ticket(
+            movie_session=data.get("movie_session"),
+            row=data.get("row"),
+            seat=data.get("seat"),
+        )
+        ticket.full_clean()
+        return data
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=True, allow_null=False)
```

Comment:
The `tickets` field is marked as `read_only=True`, which means it will not be included in the `validated_data` during creation. Therefore, attempting to `pop` it in the `create` method will raise a KeyError. Consider removing the `tickets` field from `validated_data` handling or adjusting the logic to accommodate read-only fields.